src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -740,3 +740,77 @@ impl std::cmp::PartialOrd for DatasetRefAny {
         Some(self.cmp(other))
     }
 }
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum DatasetRefPattern {
+    Ref(DatasetRef),
+    Pattern(Option<AccountName>, DatasetNamePattern),
+}
+
+impl DatasetRefPattern {
+    pub fn is_match(&self, dataset_hande: &DatasetHandle) -> bool {
+        match self {
+            Self::Ref(dataset_ref) => match dataset_ref {
+                DatasetRef::ID(dataset_id) => *dataset_id == dataset_hande.id,
+                DatasetRef::Alias(dataset_alias) => *dataset_alias == dataset_hande.alias,
+                DatasetRef::Handle(dataset_handle_res) => dataset_handle_res == dataset_hande,
+            },
+            Self::Pattern(account_name_maybe, dataset_name_pattern) => {
+                (account_name_maybe.is_none()
+                    || *account_name_maybe == dataset_hande.alias.account_name)
+                    && dataset_name_pattern.is_match(&dataset_hande.alias.dataset_name)
+            }
+        }
+    }
+
+    pub fn dataset_ref(&self) -> Option<&DatasetRef> {
```

Comment:
`as_dataset_ref` might be a better name to underline that we're not accessing some field, but re-interpreting this instance as an enum variant.